sbor/src/schema/type_link.rs

```diff
@@ -0,0 +1,111 @@
+use sbor::*;
+
+/// Marker trait for a link between [`TypeKind`]s:
+/// - [`GlobalTypeId`]: A global identifier for a type (a well known id, or type hash)
+/// - [`LocalTypeIndex`]: A link in the context of a schema (a well known id, or a local index)
+pub trait SchemaTypeLink: Clone + PartialEq + Eq {}
+
+/// This is a global identifier for a type.
+#[derive(Clone, Copy, Debug, Hash, PartialEq, Eq, Encode, Decode, TypeId)]
+pub enum GlobalTypeId {
+    /// This takes a well_known type index.
+    ///
+    /// This wraps a `[u8; 1]` because it needs to be able to be turned into a `[u8]` in a const context.
+    WellKnown([u8; 1]),
+    /// The global type hash of a type - used for types which aren't well known.
+    Novel(TypeHash),
```

Comment:
`Novel` is fun.

Reply:
Glad you like it. I originally had `Custom` but that was used to mean other things. So I did synonym lookup and liked the term `Novel` 👍